xarray/core/computation.py

```diff
@@ -1860,7 +1860,7 @@ def where(cond, x, y, keep_attrs=None):
     if keep_attrs is True:
         # keep the attributes of x, the second parameter, by default to
         # be consistent with the `where` method of `DataArray` and `Dataset`
-        keep_attrs = lambda attrs, context: getattr(x, "attrs", {})
+        keep_attrs = lambda attrs, context: attrs[1] if len(attrs) > 1 else {}
```

Comment:
Should we factor this out to a helper as [suggested](https://github.com/pydata/xarray/pull/6461#discussion_r1005023395) by @keewis?